packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,290 @@
+            fromAdapterEvent('ready'),
+            tap(() => this.patchState({ ready: true }))
+        );
+    });
+
+    readonly onError = this.effect(() => {
+        return this.adapter$.pipe(
+            isNotNull,
+            fromAdapterEvent('error'),
+            tap((error) => this.logError(error))
+        );
+    });
+
+    private logError = (error: unknown) => console.error(error);
+
+    setErrorHandler(errorHandler: (error: unknown) => void) {
+        this.logError = errorHandler;
+    }
+
+    sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options?: SendTransactionOptions
+    ): Observable<string> {
+        return this.state$.pipe(
+            first(),
+            concatMap(({ adapter, connected }) => {
+                if (!adapter) {
+                    return throwError(new WalletNotSelectedError());
+                }
+
+                if (!connected) {
+                    return throwError(new WalletNotConnectedError());
+                }
+
+                return from(defer(() => adapter.sendTransaction(transaction, connection, options))).pipe(
+                    map((txId) => txId as string),
+                    catchError(() => EMPTY)
+                );
+            })
+        );
+    }
+
+    signTransaction(transaction: Transaction): Observable<Transaction> {
+        return this.state$.pipe(
+            first(),
+            concatMap(({ adapter, connected }) => {
+                if (!adapter) {
+                    return throwError(new WalletNotSelectedError());
+                }
+
+                if (!connected) {
+                    return throwError(new WalletNotConnectedError());
+                }
+
+                if (!('signTransaction' in adapter)) {
+                    return throwError(new SignTransactionNotFoundError());
+                }
+
+                return from(defer(() => adapter.signTransaction(transaction))).pipe(
+                    map((transaction) => transaction as Transaction)
+                );
+            })
+        );
+    }
+
+    signAllTransactions(transactions: Transaction[]): Observable<Transaction[]> {
+        return this.state$.pipe(
+            first(),
+            concatMap(({ adapter, connected }) => {
+                if (!adapter) {
+                    return throwError(new WalletNotSelectedError());
+                }
+
+                if (!connected) {
+                    return throwError(new WalletNotConnectedError());
+                }
+
+                if (!('signAllTransactions' in adapter)) {
+                    return throwError(new SignAllTransactionsNotFoundError());
+                }
+
+                return from(defer(() => adapter.signAllTransactions(transactions))).pipe(
+                    map((transactions) => transactions as Transaction[])
+                );
+            })
+        );
+    }
+
+    signMessage(message: Uint8Array): Observable<Uint8Array> {
```

Comment:
Touched on this already, but my thinking is that these methods should be conditionally defined. Right now, Phantom is the only wallet that supports arbitrary message signing. It's a rough UX if users get hit with a runtime error because it's not supported by their wallet and the dapp doesn't have a reliably way of detecting this.